typescript/packages/lookslike-high-level/src/recipes/coder.ts

```diff
@@ -0,0 +1,87 @@
+import { html } from "@commontools/common-html";
+import { recipe, UI, NAME, handler, } from "@commontools/common-builder";
+import ts from 'typescript';
+import * as commonHtml from "@commontools/common-html";
+import * as commonBuilder from "@commontools/common-builder";
+import * as commonRunner from "@commontools/common-runner";
+import * as commonData from "../data.js";
+
+const runCode = handler<{}, { src: string, output: string }>(
+    (_, state) => {
+        // Custom module resolution
+        const customRequire = (moduleName: string) => {
+            switch (moduleName) {
+                case "@commontools/common-html":
+                    return commonHtml;
+                case "@commontools/common-builder":
+                    return commonBuilder;
+                case "@commontools/common-runner":
+                    return commonRunner;
+                case "../data.js":
+                    return commonData;
+                default:
+                    throw new Error(`Module not found: ${moduleName}`);
+            }
+        };
+
+        const result = ts.transpileModule(state.src, {
+            compilerOptions: {
+                module: ts.ModuleKind.CommonJS,
+                target: ts.ScriptTarget.ES2015,
+            }
+        });
+
+        const logs: string[] = [];
+        const originalLog = console.log;
+        console.log = (...args) => {
```

Comment:
nit, but adding a `console` variable in the scope around the eval doesn't work? (unless the code says `window.console.log`, but who does that :) )